src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,151 @@
+import classNames from 'classnames';
 import React from 'react';
+import { getTodos } from '../../api/api';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+type State = {
+  todos: Todo[];
+  titleSearch: string;
+  statusOfTodos: string;
+};
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+type Props = {
+  selectedUserId: number;
+  selectUser: (userId: number) => void;
+};
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    titleSearch: '',
+    statusOfTodos: 'any',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({
+      todos,
+    });
+  }
+
+  handleChangeTodoStatus = (todoId: number) => {
+    const { todos } = this.state;
+
+    const updatedTodos = todos.map(todo => {
+      if (todo.id === todoId) {
+        return { ...todo, completed: !todo.completed };
+      }
+
+      return todo;
+    });
+
+    this.setState({
+      todos: updatedTodos,
+    });
+  };
+
+  handleChangeSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ titleSearch: e.currentTarget.value });
+  };
+
+  handleChangeSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ statusOfTodos: e.currentTarget.value });
+  };
 
-          <button
-            className="TodoList__user-button button"
-            type="button"
+  filterTodos = (): Todo[] => {
+    const { todos, statusOfTodos, titleSearch } = this.state;
+    let filteredTodos;
+
+    switch (statusOfTodos) {
+      case 'active':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+
+      default:
+        filteredTodos = [...todos];
+    }
+
+    return filteredTodos.filter(todo => (
+      todo.title.toLowerCase().includes(titleSearch.toLocaleLowerCase())
+    ));
```

Comment:
titleSearch.toLowerCase() - better to write in variable, for each filter iteration you make operation toLowerCase for titleSearch
with the separate variable, you will make toLowerCase only once